ValheimPlus/GameClasses/Smelter.cs

```diff
@@ -151,102 +151,99 @@ static void Prefix(Smelter __instance)
+                ItemDrop.ItemData fuelItemData = __instance.m_fuelItem.m_itemData;
 
                 // Check for fuel in nearby containers
-                int addedFuel = InventoryAssistant.RemoveItemInAmountFromAllNearbyChests(smelter.gameObject, autoFuelRange, fuelItemData, toMaxFuel, !ignorePrivateAreaCheck);
+                int addedFuel = InventoryAssistant.RemoveItemInAmountFromAllNearbyChests(__instance.gameObject, autoFuelRange, fuelItemData, toMaxFuel, !ignorePrivateAreaCheck);
                 for (int i = 0; i < addedFuel; i++)
                 {
-                    smelter.m_nview.InvokeRPC("AddFuel", new object[] { });
+                    __instance.m_nview.InvokeRPC("AddFuel", new object[] { });
                 }
                 if (addedFuel > 0)
-                    ZLog.Log("Added " + addedFuel + " fuel(" + fuelItemData.m_shared.m_name + ") in " + smelter.m_name);
+                    UnityEngine.Debug.Log("Added " + addedFuel + " fuel(" + fuelItemData.m_shared.m_name + ") in " + __instance.m_name);
             }
             if (toMaxOre > 0)
             {
-                List<Container> nearbyChests = InventoryAssistant.GetNearbyChests(smelter.gameObject, autoFuelRange);
-                foreach (Container c in nearbyChests)
+                foreach (Smelter.ItemConversion itemConversion in __instance.m_conversion)
                 {
-                    foreach (Smelter.ItemConversion itemConversion in smelter.m_conversion)
+                    if (isKiln)
                     {
-                        if (isKiln)
-                        {
-                            if (Configuration.Current.Kiln.dontProcessFineWood && itemConversion.m_from.m_itemData.m_shared.m_name.Equals(WoodDefinitions.FineWoodName)) continue;
-                            if (Configuration.Current.Kiln.dontProcessRoundLog && itemConversion.m_from.m_itemData.m_shared.m_name.Equals(WoodDefinitions.RoundLogName)) continue;
+                        if (Configuration.Current.Kiln.dontProcessFineWood && itemConversion.m_from.m_itemData.m_shared.m_name.Equals(WoodDefinitions.FineWoodName)) continue;
+                        if (Configuration.Current.Kiln.dontProcessRoundLog && itemConversion.m_from.m_itemData.m_shared.m_name.Equals(WoodDefinitions.RoundLogName)) continue;
 
-                            int threshold = Configuration.Current.Kiln.stopAutoFuelThreshold < 0 ? 0 : Configuration.Current.Kiln.stopAutoFuelThreshold;
-                            if (threshold > 0 && InventoryAssistant.GetItemAmountInItemList(InventoryAssistant.GetNearbyChestItemsByContainerList(nearbyChests), itemConversion.m_to.m_itemData) >= threshold) return;
-                        }
+                        if (threshold > 0 && InventoryAssistant.GetItemAmountInItemList(
+                            InventoryAssistant.GetNearbyChestItems(__instance.gameObject, autoFuelRange, !ignorePrivateAreaCheck), itemConversion.m_to.m_itemData) >= threshold
```

Comment:
You should move the `InventoryAssistant.GetNearbyChestItems` call out of the foreach loop to avoid calling multiple time this function.

This function calls `InventoryAssistant.GetNearbyChests` which uses `Physics.OverlapSphere`. This is a function that has an heavy impact on performance and should only be run once for a given `__instance.gameObject`.